src/vos/vos_io.c

```diff
@@ -2617,12 +2623,11 @@ vos_update_begin(daos_handle_t coh, daos_unit_oid_t oid, daos_epoch_t epoch,
 
 	/* Hold the object for the evictable md-on-ssd phase2 pool */
 	if (vos_pool_is_evictable(vos_cont2pool(ioc->ic_cont))) {
-		rc = vos_ts_set_add(ioc->ic_ts_set, ioc->ic_cont->vc_ts_idx, NULL, 0);
-		D_ASSERT(rc == 0);
-
-		rc = vos_obj_hold(ioc->ic_cont, ioc->ic_oid, &ioc->ic_epr, ioc->ic_bound,
-				  VOS_OBJ_VISIBLE | VOS_OBJ_CREATE, DAOS_INTENT_UPDATE,
-				  &ioc->ic_obj, ioc->ic_ts_set);
+		/*
+		 * To avoid the complication of adding object ilog to ts_set, we simply pin object
+		 * here without ts_set involved.
+		 */
+		rc = vos_obj_acquire(ioc->ic_cont, ioc->ic_oid, true, &ioc->ic_pinned_obj);
```

Comment:
It seems that the `ioc->ic_pinned_obj` and `ioc->ic_obj` are for the same OID (`ioc->ic_oid`)?